pkg/solana/config/chain_reader_test.go

```diff
@@ -0,0 +1,117 @@
+package config_test
+
+import (
+	_ "embed"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	codeccommon "github.com/smartcontractkit/chainlink-common/pkg/codec"
+	"github.com/smartcontractkit/chainlink-common/pkg/codec/encodings/binary"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/codec/testutils"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/config"
+)
+
+//go:embed testChainReader_valid.json
+var validJSON string
+
+//go:embed testChainReader_invalid.json
+var invalidJSON string
+
+func TestChainReaderConfig(t *testing.T) {
+	t.Parallel()
+
+	t.Run("valid unmarshal", func(t *testing.T) {
+		t.Parallel()
+
+		var result config.ChainReader
+		require.NoError(t, json.Unmarshal([]byte(validJSON), &result))
+		assert.Equal(t, validChainReaderConfig, result)
+	})
+
+	t.Run("invalid unmarshal", func(t *testing.T) {
+		t.Parallel()
+
+		var result config.ChainReader
+		require.NotNil(t, json.Unmarshal([]byte(invalidJSON), &result))
```

Comment:
I was staring at this thinking "how can this be working with `require.NotNil`?" then realized this is the line the CI failure happened on!

```suggestion
		_, err = json.Unmarshal([]byte(invalidJSON), &result))
		assert.NoError(t, err)
```

Reply:
(or `assert.ErrorIs` with whatever error type it's supposed to return?)